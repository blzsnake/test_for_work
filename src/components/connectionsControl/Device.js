import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Text, Button } from 'ghofi-react-components';

import s from './styles.less';

import mobile from './images/mobile.svg';
import tablet from './images/tab.svg';
import router from './images/router.svg';
import tv from './images/tv.svg';
import pc from './images/pc.svg';
import mobileDisabled from './images/mobile-disabled.svg';
import tabDisabled from './images/tab-disabled.svg';
import routerDisabled from './images/router-disabled.svg';
import tvDisabled from './images/tv-disabled.svg';
import pcDisabled from './images/pc-disabled.svg';

const TYPES_ICO = {
  mobile,
  tablet,
  tv,
  pc,
  notebook: pc,
  router,
  mobileDisabled,
  tabDisabled,
  routerDisabled,
  tvDisabled,
  pcDisabled,
};

/**
 * @param  {string} {name=''
 * @param  {string} type=''
 * @param  {number} pin=0
 * @param  {string} mac=0
 * @param  {bool} disabled}
 */

const Device = ({ name = 'name', type = 'mobile', pin = 0, label = '', mac = '', disabled, handleClick, handleCancelPinClick, id }) => (
  <div className={s.device}>
    <div
      role='button'
      className={s.device__view}
      onKeyDown={handleClick({ pin, mac, name, type, label, client_id: id }, true)}
      onClick={handleClick({ pin, mac, name, type, label, client_id: id }, true)}
      tabIndex={0}>
      <img
        alt={type}
        src={disabled ? TYPES_ICO[`${type}Disabled`] : TYPES_ICO[type]}
        className={s.device__img} />
    </div>
    <div className={s.device__info}>
      <div className={s.device__name}>
        <Text
          size={16}
          sizeTablet={14}
          fontInherit>{name}
        </Text>
      </div>
      {!disabled &&
      <div className={s.device__pin}>
        <Text
          size={16}
          sizeTablet={14}
          fontInherit>{`PIN ${pin}`}
        </Text>
      </div>}
      {!disabled &&
      <div className={s.device__button}>
        <Button
          wide
          onClick={handleCancelPinClick({ pin, mac })}
          mod='device-cancel'
          color='yellow'>
          <Text
            size={11}
            uppercase
            fontInherit>
            <FormattedMessage id='connections_control_button' />
          </Text>
        </Button>
      </div>}
    </div>
  </div>
);

export default Device;
