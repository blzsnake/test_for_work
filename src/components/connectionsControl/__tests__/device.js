import React from 'react';
import { shallow } from 'enzyme';

import Device from '../Device';

const props = {
  freePinQuantity: 11,
  name: 'Mark Yuliy Cesar',
  type: 'string',
  pin: 1,
  mac: 1,
  handleClick: () => {},
  handleCancelPinClick: () => {},
};

describe('Should render', () => {
  it('Device render', () => {
    const component = shallow(<Device {...props} />);

    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
