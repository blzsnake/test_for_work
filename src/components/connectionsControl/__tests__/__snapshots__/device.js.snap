// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Device render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Device
    freePinQuantity={11}
    handleCancelPinClick={[Function]}
    handleClick={[Function]}
    mac={1}
    name="Mark Yuliy Cesar"
    pin={1}
    type="string"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
          onClick={undefined}
          onKeyDown={undefined}
          role="button"
          tabIndex={0}
        >
          <img
            alt="string"
            className={undefined}
            src={undefined}
          />
        </div>,
        <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            <Styled(Text)
              fontInherit={true}
              size={16}
              sizeTablet={14}
            >
              Mark Yuliy Cesar
            </Styled(Text)>
          </div>
          <div
            className={undefined}
          >
            <Styled(Text)
              fontInherit={true}
              size={16}
              sizeTablet={14}
            >
              PIN 1
            </Styled(Text)>
          </div>
          <div
            className={undefined}
          >
            <Styled(Button)
              color="yellow"
              mod="device-cancel"
              onClick={undefined}
              wide={true}
            >
              <Styled(Text)
                fontInherit={true}
                size={11}
                uppercase={true}
              >
                <FormattedMessage
                  id="connections_control_button"
                  values={Object {}}
                />
              </Styled(Text)>
            </Styled(Button)>
          </div>
        </div>,
      ],
      "className": "device",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="string"
            className={undefined}
            src={undefined}
          />,
          "className": undefined,
          "onClick": undefined,
          "onKeyDown": undefined,
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "string",
            "className": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
            >
              <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                Mark Yuliy Cesar
              </Styled(Text)>
            </div>,
            <div
              className={undefined}
            >
              <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                PIN 1
              </Styled(Text)>
            </div>,
            <div
              className={undefined}
            >
              <Styled(Button)
                color="yellow"
                mod="device-cancel"
                onClick={undefined}
                wide={true}
              >
                <Styled(Text)
                  fontInherit={true}
                  size={11}
                  uppercase={true}
                >
                  <FormattedMessage
                    id="connections_control_button"
                    values={Object {}}
                  />
                </Styled(Text)>
              </Styled(Button)>
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                Mark Yuliy Cesar
              </Styled(Text)>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Mark Yuliy Cesar",
                "fontInherit": true,
                "size": 16,
                "sizeTablet": 14,
              },
              "ref": null,
              "rendered": "Mark Yuliy Cesar",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                PIN 1
              </Styled(Text)>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PIN 1",
                "fontInherit": true,
                "size": 16,
                "sizeTablet": 14,
              },
              "ref": null,
              "rendered": "PIN 1",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Styled(Button)
                color="yellow"
                mod="device-cancel"
                onClick={undefined}
                wide={true}
              >
                <Styled(Text)
                  fontInherit={true}
                  size={11}
                  uppercase={true}
                >
                  <FormattedMessage
                    id="connections_control_button"
                    values={Object {}}
                  />
                </Styled(Text)>
              </Styled(Button)>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Text)
                  fontInherit={true}
                  size={11}
                  uppercase={true}
                >
                  <FormattedMessage
                    id="connections_control_button"
                    values={Object {}}
                  />
                </Styled(Text)>,
                "color": "yellow",
                "mod": "device-cancel",
                "onClick": undefined,
                "wide": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <FormattedMessage
                    id="connections_control_button"
                    values={Object {}}
                  />,
                  "fontInherit": true,
                  "size": 11,
                  "uppercase": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "connections_control_button",
                    "values": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
            onClick={undefined}
            onKeyDown={undefined}
            role="button"
            tabIndex={0}
          >
            <img
              alt="string"
              className={undefined}
              src={undefined}
            />
          </div>,
          <div
            className={undefined}
          >
            <div
              className={undefined}
            >
              <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                Mark Yuliy Cesar
              </Styled(Text)>
            </div>
            <div
              className={undefined}
            >
              <Styled(Text)
                fontInherit={true}
                size={16}
                sizeTablet={14}
              >
                PIN 1
              </Styled(Text)>
            </div>
            <div
              className={undefined}
            >
              <Styled(Button)
                color="yellow"
                mod="device-cancel"
                onClick={undefined}
                wide={true}
              >
                <Styled(Text)
                  fontInherit={true}
                  size={11}
                  uppercase={true}
                >
                  <FormattedMessage
                    id="connections_control_button"
                    values={Object {}}
                  />
                </Styled(Text)>
              </Styled(Button)>
            </div>
          </div>,
        ],
        "className": "device",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="string"
              className={undefined}
              src={undefined}
            />,
            "className": undefined,
            "onClick": undefined,
            "onKeyDown": undefined,
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "string",
              "className": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
              >
                <Styled(Text)
                  fontInherit={true}
                  size={16}
                  sizeTablet={14}
                >
                  Mark Yuliy Cesar
                </Styled(Text)>
              </div>,
              <div
                className={undefined}
              >
                <Styled(Text)
                  fontInherit={true}
                  size={16}
                  sizeTablet={14}
                >
                  PIN 1
                </Styled(Text)>
              </div>,
              <div
                className={undefined}
              >
                <Styled(Button)
                  color="yellow"
                  mod="device-cancel"
                  onClick={undefined}
                  wide={true}
                >
                  <Styled(Text)
                    fontInherit={true}
                    size={11}
                    uppercase={true}
                  >
                    <FormattedMessage
                      id="connections_control_button"
                      values={Object {}}
                    />
                  </Styled(Text)>
                </Styled(Button)>
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Styled(Text)
                  fontInherit={true}
                  size={16}
                  sizeTablet={14}
                >
                  Mark Yuliy Cesar
                </Styled(Text)>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Mark Yuliy Cesar",
                  "fontInherit": true,
                  "size": 16,
                  "sizeTablet": 14,
                },
                "ref": null,
                "rendered": "Mark Yuliy Cesar",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Styled(Text)
                  fontInherit={true}
                  size={16}
                  sizeTablet={14}
                >
                  PIN 1
                </Styled(Text)>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "PIN 1",
                  "fontInherit": true,
                  "size": 16,
                  "sizeTablet": 14,
                },
                "ref": null,
                "rendered": "PIN 1",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Styled(Button)
                  color="yellow"
                  mod="device-cancel"
                  onClick={undefined}
                  wide={true}
                >
                  <Styled(Text)
                    fontInherit={true}
                    size={11}
                    uppercase={true}
                  >
                    <FormattedMessage
                      id="connections_control_button"
                      values={Object {}}
                    />
                  </Styled(Text)>
                </Styled(Button)>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Text)
                    fontInherit={true}
                    size={11}
                    uppercase={true}
                  >
                    <FormattedMessage
                      id="connections_control_button"
                      values={Object {}}
                    />
                  </Styled(Text)>,
                  "color": "yellow",
                  "mod": "device-cancel",
                  "onClick": undefined,
                  "wide": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <FormattedMessage
                      id="connections_control_button"
                      values={Object {}}
                    />,
                    "fontInherit": true,
                    "size": 11,
                    "uppercase": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "connections_control_button",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
